v0/sources/appsflyer/transform.js

```diff
@@ -0,0 +1,68 @@
+const path = require("path");
+const fs = require("fs");
+const Message = require("../message");
+
+const mappingJson = JSON.parse(
+  fs.readFileSync(path.resolve(__dirname, "./mapping.json"), "utf-8")
+);
+
+const { removeUndefinedAndNullValues } = require("../../util");
+
+function processEvent(event) {
+  const messageType = "track";
+
+  if (event.event_name) {
+    const eventName = event.event_name;
+    const message = new Message(`AF`);
+
+    message.setEventType(messageType);
+
+    message.setEventName(eventName);
+
+    const properties = { ...event };
+    message.setProperty("properties", properties);
+
+    // set fields in payload from mapping json
+    message.setProperties(event, mappingJson);
+
+    // Remove the fields from properties that are already mapped to other fields.
+    Object.keys(mappingJson).forEach(key => {
```

Comment:
Removing the mapped properties earlier, it's not handling the keys `idfa`, `appsflyer_id` etc.